src/data/zmod/basic.lean

```diff
@@ -261,14 +273,12 @@ def units_equiv_coprime {n : ℕ+} : units (zmod n) ≃ {x : zmod n // nat.copri
       units.coe_mul, zmod.mul_val, zmod.cast_mod_nat, zmod.cast_mod_nat,
       zmod.eq_iff_modeq_nat] at this
     end⟩,
-  inv_fun := λ x,
-    have x.val * ↑(gcd_a ((x.val).val) ↑n) = 1,
-      by rw [← zmod.cast_val x.1, ← int.cast_coe_nat, ← int.cast_one, ← int.cast_mul,
-          zmod.eq_iff_modeq_int, ← int.coe_nat_one, ← (show nat.gcd _ _ = _, from x.2)];
-        simpa using int.modeq.gcd_a_modeq x.1.1 n,
-    ⟨x.1, gcd_a x.1.1 n, this, by simpa [mul_comm] using this⟩,
-  left_inv := λ ⟨_, _, _, _⟩, units.ext rfl,
-  right_inv := λ ⟨_, _⟩, rfl }
+  inv_fun := λ x, unit_of_coprime x.1.1 x.2,
+  left_inv := λ ⟨_, _, _, _⟩, units.ext (by simp),
+  right_inv := λ ⟨_, _⟩, by simp }
+
+instance units.fintype {n : ℕ+} : fintype (units (zmod n)) :=
```

Comment:
Shouldn't this be generalised? `zmod n` is a fintype, hence `units` is a fintype.

Reply:
We are moving away from constructivism by defining things in that generality, but I'm okay with that.

Reply:
Hmm, I'm certainly ok with that.
But I would expect that in a finite ring you can constructively build the fintype of units. Anyway, I don't really care.

Reply:
With a `decidable_eq` assumption it can be made computable by just searching for the inverse, but this is getting towards pointless computability I think, where the most general algorithm is so slow it's kind of pointless, so I just made a noncomputable instance.